src/prompts/checkly-data.ts

```diff
@@ -0,0 +1,71 @@
+import { ChecklyClient } from "../checkly/checklyclient.ts";
+import { CheckResult } from "../checkly/models.ts";
+
+// export async function fetchChecksForGroup(checks: Check[]) {
+//   const intervalStart = Date.now() - 30 * 24 * 60 * 60 * 1000;
+//   const intervalEnd = Date.now();
+
+//   const results: any[] = [];
+//   for (const check of checks) {
+//     // Get failed results from last 30 days
+
+//     return results;
+//   }
+// }
+
+export const LAST_30_DAYS = {
+  from: Date.now() - 30 * 24 * 60 * 60 * 1000,
+  to: Date.now(),
+};
+
+export const LAST_24_HOURS = {
+  from: Date.now() - 24 * 60 * 60 * 1000,
+  to: Date.now(),
+};
+
+export const LAST_1_HOURS = {
+  from: Date.now() - 60 * 60 * 1000,
+  to: Date.now(),
+};
+
+export const last24h = (date: Date) => {
+  return {
+    from: date.getTime() - 24 * 60 * 60 * 1000,
+    to: date.getTime(),
+  };
+};
+
+export async function fetchCheckResults(
+  checkly: ChecklyClient,
+  {
+    checkId,
+    from,
+    to,
+  }: {
+    checkId: string;
+    from?: number;
+    to?: number;
+  },
+) {
+  return await checkly.getCheckResultsByCheckId(checkId, {
+    resultType: "ALL",
+    from: from ?? Date.now() - 30 * 24 * 60 * 60 * 1000,
+    to: to ?? Date.now(),
+    limit: 100,
+  });
+}
+
+export function summarizeCheckResult(checkResult: CheckResult) {
+  const error =
+    checkResult.browserCheckResult?.errors.find((e) => !!e.message)?.message ||
```

Comment:
make it work for API and Multistep checks also?

Reply:
out of scope for now. Lets do this in a new mr with a dedicated ticket.

Reply:
https://linear.app/checklyhq/issue/AIT-31/analysis-of-individual-errors-for-api-and-multistep-checks